awscli/customizations/ec2/paginate.py

```diff
@@ -14,44 +14,27 @@
 
 class EC2PageSizeInjector(object):
 
-    DEFAULT_TARGET_OPERATIONS = {
+    # Operations to auto-paginate and their specific whitelists.
+    # Format:
+    #    Key:   Operation
+    #    Value: List of parameters to add to whitelist for that operation.
+    TARGET_OPERATIONS = {
         "describe-volumes": [],
         "describe-snapshots": ['owner_ids', 'restorable_by_user_ids']
     }
 
-    DEFAULT_GLOBAL_WHITELIST = [
+    # Parameters which should be whitelisted for every operation.
+    GLOBAL_WHITELIST = [
         'cli_input_json', 'generate_cli_skeleton', 'help', 'max_items',
         'next_token', 'dry_run', 'starting_token'
     ]
 
-    def __init__(self, default_page_size=1000, target_operations=None,
-                 global_whitelist=None):
-        """
-        :type default_page_size: int
-        :param default_page_size: The value to inject for page_size.
-
-        :type target_operations: dict
-        :param target_operations: A dictionary whose keys are operations and
-            whose values are parameters that are white-listed for that
-            operation. If a parameter is specified that is not in that list,
-            page_size will not be set.
-
-        :type global_whitelist: list of str
-        :param global_whitelist: A list of parameters that should be
-            whitelisted for every operation.
-        """
-        self._default_page_size = default_page_size
-        self._target_operations = target_operations
-        if self._target_operations is None:
-            self._target_operations = self.DEFAULT_TARGET_OPERATIONS
-        self._global_whitelist = global_whitelist
-        if self._global_whitelist is None:
-            self._global_whitelist = self.DEFAULT_GLOBAL_WHITELIST
+    DEFAULT_PAGE_SIZE = 1000
```

Comment:
Oh I was fine with the ``page_size`` constructor parameter as it would make sense to individually tune each injector, but I am fine with it being a class variable for now and refactoring it if we need to choose different defaults for different operations